Keep rosed from running if the file doesn't exist in the package

timonegk opened the pull request:
Same as abb02a22d4c67638dcc95a80241ade95658a3b02, but for zsh.

timonegk committed:
Keep rosed from running if the file doesn't exist in the package

Same as abb02a22d4c67638dcc95a80241ade95658a3b02, but for zsh.

hidmic reviewed (approved):
LGTM, @sloretz any thoughts?

jacobperron reviewed (commented):
Sounds reasonable to keep the behavior consistent, though I'm not sure if we want to break behavior mid-distro.

hidmic commented:
> though I'm not sure if we want to break behavior mid-distro.

Fair point. I don't have a strong opinion. It is somewhat surprising for the same tool to behave differently for each shell though.

timonegk committed:
Unify behavior of rosed, roscat, roscd, rosmv for unknown packages

timonegk commented:
I just saw the check is done for `roscat` (using `$?` and `return`) and thought that would be better for the other tools, too.

timonegk commented:
Also, I don't think that any of the changes is actually a breaking change. `rosed` with an unknown package name currently just opens an empty editor without any file selected, I can't imagine any use case for that. Also, `rosed` is not a tool that can be used in scripts.
`roscp` and `rosmv` currently fail, but with the following message:
```
Couldn't find package [foo]
cp: missing destination file operand after '/path/to/destination'
Try 'cp --help' for more information.
```
(and the same for `mv`)
This is changed to just the first line (`Couldn't find package [foo]`), which is less confusing.

timonegk commented:
While we're at it, I just applied the unification to bash as well.

hidmic reviewed (approved):
As I said above, I'm onboard with these changes. Regarding:

> rosed with an unknown package name currently just opens an empty editor without any file selected, I can't imagine any use case for that. Also, rosed is not a tool that can be used in scripts.

@timonegk why not adjusting `rosed` to skip that `_roscmd` call if `$#` is below 1 and just open the editor? That way we don't break any use case (no matter how odd it may be).

timonegk committed:
open empty editor when rosed is called without arguments

timonegk commented:
Sure, I updated the files.
Before you ask: I removed the quotes in rosbash because then, the editor can be called when $arg is empty. When it is called with "", the behavior is different. This might break when there are spaces in the file name, but that already breaks in _roscmd and should be fixed there, which is a different issue.

hidmic commented:
>  I removed the quotes in rosbash because then, the editor can be called when $arg is empty. When it is called with "", the behavior is different. This might break when there are spaces in the file name, but that already breaks in _roscmd and should be fixed there, which is a different issue.

Ah, that's not ideal, but OK. @jacobperron ?

hidmic commented:
@jacobperron any thoughts against merging?

jacobperron commented:
@hidmic Sounds okay to me. Sorry for the delay.
